Blizzard.eai

```diff
@@ -1,1760 +1,1985 @@
+    call TriggerRegisterDialogEvent(t, start_dialog )
+    set last_dialog_trigger_action = TriggerAddAction(t, function StartDialogResponse)
+    set last_dialog_trigger = t
+    call DialogDisplay(GetTriggerPlayer(), start_dialog, true)
+  endif
+  if (b == clanguage_button) then
+    set clanguage_dialog = BuildCommanderLanguage()
+    set language_trigger = CreateTrigger()
+    call TriggerRegisterDialogEvent(language_trigger, clanguage_dialog)
+    call TriggerAddAction(language_trigger, function CommanderLanguageResponse)
+    call DialogDisplay(GetTriggerPlayer(), clanguage_dialog, true)
+  endif
+  if (b == all_button or sdbn_button[i] == b) then
+    if (sdbn_button[i] == b) then
+      if GetPlayerRace(Player(sdbn_number[i])) == RACE_HUMAN then
+        set RaceStrategy = "HUMAN"
+      elseif GetPlayerRace(Player(sdbn_number[i])) == RACE_ORC then
+        set RaceStrategy = "ORC"
+      elseif GetPlayerRace(Player(sdbn_number[i])) == RACE_UNDEAD then
+        set RaceStrategy = "UNDEAD"
+      elseif GetPlayerRace(Player(sdbn_number[i])) == RACE_NIGHTELF then
+        set RaceStrategy = "ELF"
+      endif
+      set commanded_player[GetPlayerId(GetTriggerPlayer())] = Player(sdbn_number[i])
+    elseif (b == all_button) then
+      set commanded_player[GetPlayerId(GetTriggerPlayer())] = Player(PLAYER_NEUTRAL_AGGRESSIVE)
+    endif
+    if commanded_player[GetPlayerId(GetTriggerPlayer())] != Player(PLAYER_NEUTRAL_AGGRESSIVE) and (not IsPlayerAlly(GetTriggerPlayer(), commanded_player[GetPlayerId(GetTriggerPlayer())])) then
+      call DisplayFromDlg(chat_no_ally)
+      set b = null
+      set t = null
+      return
+    endif
+    call DialogSetMessage(root_dialog, dlghdr_root)
+    call DialogDisplay(GetTriggerPlayer(), root_dialog, true)
+  endif
+  set b = null
+  set t = null
+endfunction
+
+// Builds a pageable number dialog
+function BuildNumberDialog takes player p, integer startnum returns dialog
+  local integer maxCount = 25  // Configure this to increase total number options to display
+  local integer i = startnum
+  local dialog new_dialog = DialogCreate()
+  //call DisplayToAll("STRAT Max: " + I2S(maxCount) + " for player: " + I2S(GetPlayerId(p)) + " starting page at:" + I2S(i))
+  call DestroyLastTrigger(last_number_trigger, last_number_trigger_action)
+  set last_number_trigger = null
+  set last_number_trigger_action = null
+  call DialogDestroy(last_number_dialog)
+  set last_number_dialog = new_dialog
+  call DialogSetMessage(new_dialog, dlghdr_choose_number)
+  set ndbn_length = startnum
+  set ndbn_paged_length = 0
+  if Strategydialog == 0 or RaceStrategy == null then
+    loop
+      exitwhen i >= 9 + startnum or i >= maxCount
+      set ndbn_button[ndbn_length] = DialogAddButton(new_dialog, I2S(i), IMinBJ(48 + i, 57))
+      set ndbn_length = ndbn_length + 1
+      set ndbn_paged_length = ndbn_paged_length + 1
+      set i = i + 1
+    endloop
+  else

```

Comment:
Idea was good, reworked in this commit https://github.com/SMUnlimited/AMAI/commit/183dd69c9f800cea6658dece012b7deae1d35b8d

Reply:
For other commands, choosing numbers seems meaningless 
The num dialog seems to exist only for strategy selection , maybe other num dialog can be cancelled